src/game.cpp

```diff
@@ -5250,22 +5250,54 @@ void game::control_vehicle()
             veh->start_engines( true );
         }
     } else {
-        const cata::optional<tripoint> examp_ = choose_adjacent( _( "Control vehicle where?" ) );
-        if( !examp_ ) {
-            return;
+        std::vector<optional_vpart_position> vehicles;
+        tripoint sole_position;
+        for( const tripoint elem : m.points_in_radius( g->u.pos(), 1 ) ) {
+            if( const optional_vpart_position vp = m.veh_at( elem ) ) {
+                const cata::optional<vpart_reference> vpr = vp.value().part_with_feature( "CONTROLS", true );
+                if( vpr ) {
+                    vehicles.push_back( vp );
+                    sole_position = elem;
+                }
+            }
         }
-        const optional_vpart_position vp = m.veh_at( *examp_ );
-        if( !vp ) {
-            add_msg( _( "No vehicle there." ) );
+        if( vehicles.size() < 1 ) {
+            add_msg( _( "No vehicle controls in reach." ) );
             return;
-        }
-        veh = &vp->vehicle();
-        veh_part = vp->part_index();
-        if( veh->avail_part_with_feature( veh_part, "CONTROLS", true ) >= 0 ) {
-            if( !veh->handle_potential_theft( dynamic_cast<player &>( u ) ) ) {
+        } else if( vehicles.size() > 1 ) {
+            const cata::optional<tripoint> examp_ = choose_adjacent( _( "Control vehicle where?" ) );
+            if( !examp_ ) {
+                return;
+            }
+            optional_vpart_position vp = m.veh_at( *examp_ );
+            if( vp ) {
+                const cata::optional<vpart_reference> vpr = vp.value().part_with_feature( "CONTROLS", true );
+                if( vpr ) {
+                    veh = &vp->vehicle();
+                    veh_part = vp->part_index();
+                    if( veh->avail_part_with_feature( veh_part, "CONTROLS", true ) >= 0 ) {
+                        if( !veh->handle_potential_theft( dynamic_cast<player &>( u ) ) ) {
+                            return;
+                        }
+                        veh->use_controls( *examp_ );
+                    }
+                } else {
+                    add_msg( _( "The vehicle doesn't have controls there." ) );
+                    return;
+                }
+            } else {
+                add_msg( _( "No vehicle there." ) );
```

Comment:
This will never occur, because it only has a change of being shown if there are 2 or more vehicle parts with controls within one tile of you. I suggest you do as @NickGrease suggests and replace the message for the vehicles vector being empty with this message.

Reply:
This still can be shown, though we've found several candidates, the player can still screw up and pick an empty tile. Right?

Reply:
Doh! I missed that, sorry.